gnovm/cmd/gno/test.go

```diff
@@ -145,75 +150,72 @@ func execTest(cfg *testCfg, args []string, io *commands.IO) error {
 		}()
 	}
 
+	subPkgs, err := gnomod.SubPkgsFromPaths(paths)
```

Comment:
we CAN have a function called "SubPkgsFromDir" (singular), but not "SubPkgsFromDirs" or "SubPkgsFromPath" or "SubPkgsFromPaths".

Reply:
Based on how `go list ./crypto/... ./foo ./crypto/dsa` works (https://github.com/golang/go/blob/499a12009938fe2ffff90775832b9d67ca3e46b2/src/cmd/go/internal/list/list.go#L569C24-L569C35), they also use a single helper that takes a list of targets.

The main difference is that they are only supporting dirs and not single file, while we want our CLI to basically support this at the end: `gno test ./crypto/... ./foo ./bar_test.gno` with files too.

I think we should consider that we have:
* `PkgPath` when onchain or when identified by gno.mod or other methods
* `Dir` in the case a `PkgPath` is located in the file system (but mabe we won't need it anymore), and 
* `Target`, in my opinion target should be the only way we identify a `os.Args` that can be a pattern, a dir or a file.

---

Edit: @harry-hov found this which is better: https://github.com/golang/go/blob/master/src/cmd/go/internal/load/pkg.go#L2816; it consists of looking for packages, but creating virtual packages of a single file when a file is passed instead of a directory or pattern.